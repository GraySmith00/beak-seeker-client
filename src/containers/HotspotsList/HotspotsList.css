.hotspots-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: scroll;
  background: url("./beachBirds.jpg");
  background-size: cover;
  background-position: center center; }
  .hotspots-list .main-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 70px;
    width: 100%; }
  .hotspots-list .hotspot {
    display: flex;
    justify-content: center;
    margin: 0.5rem auto;
    padding: 0.8rem;
    width: 80%;
    background: rgba(0, 0, 0, 0.75);
    text-decoration: none;
    color: #fff;
    border-radius: 5px;
    cursor: pointer;
    -webkit-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43); }
  .hotspots-list .hotspot:last-child {
    margin-bottom: 8rem; }
