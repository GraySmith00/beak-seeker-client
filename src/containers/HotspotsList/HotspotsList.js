import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';

import { connect } from 'react-redux';

import { getNearbyHotspots } from '../../actions/thunks/getNearbyHotspots';
import LoadingSpinner from '../../components/LoadingSpinner/LoadingSpinner';

export class HotspotsList extends Component {
  componentDidMount() {
    this.props.getNearbyHotspots();
  }

  render() {
    const { loading } = this.props;
    let displayHotspotLinks;

    if (loading) {
      displayHotspotLinks = <LoadingSpinner />;
    } else {
      displayHotspotLinks = this.props.hotspots.map(hotspot => (
        <div key={hotspot.locId} className="hotspot">
          <Link to={`/hotspots/${hotspot.locId}`}>{hotspot.locName}</Link>
        </div>
      ));
    }

    return (
      <div>
        <h1>HotspotsList</h1>
        {displayHotspotLinks}
      </div>
    );
  }
}

HotspotsList.propTypes = {
  getNearbyHotspots: PropTypes.func.isRequired,
  hotspots: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
};

const mapStatetoProps = state => ({
  hotspots: state.hotspots,
  loading: state.hotspotsLoading
});

const mapDispatchToProps = dispatch => ({
  getNearbyHotspots: () => dispatch(getNearbyHotspots())
});

export default connect(
  mapStatetoProps,
  mapDispatchToProps
)(HotspotsList);
