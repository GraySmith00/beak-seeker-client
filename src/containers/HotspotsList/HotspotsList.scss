.hotspots-list {
  display: flex;
  flex-direction: column;
  align-items: center;

  .main-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    margin-top: 70px;

    width: 100%;
  }

  .hotspot {
    display: flex;
    justify-content: center;

    margin: 0.5rem auto;
    padding: 0.8rem;

    width: 80%;
    background: crimson;
    text-decoration: none;
    color: #fff;
    border-radius: 5px;

    -webkit-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    -moz-box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
    box-shadow: 10px 10px 26px -9px rgba(0, 0, 0, 0.43);
  }
}
