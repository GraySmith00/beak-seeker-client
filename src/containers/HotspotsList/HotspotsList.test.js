import React from 'react';
import { shallow } from 'enzyme';

import { HotspotsList } from './HotspotsList';
import { mockHotspots } from './mockHotspotsListData';

describe('HotspotsList component', () => {
  let wrapper;
  let mockGetNearbyHotspots;

  beforeEach(() => {
    mockGetNearbyHotspots = jest.fn();

    wrapper = shallow(
      <HotspotsList
        getNearbyHotspots={mockGetNearbyHotspots}
        hotspots={mockHotspots}
        loading={false}
      />
    );
  });

  it('should match the snapshot when the data is still loading', () => {
    wrapper = shallow(
      <HotspotsList
        getNearbyHotspots={mockGetNearbyHotspots}
        hotspots={[]}
        loading={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when the data is has been populated', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
