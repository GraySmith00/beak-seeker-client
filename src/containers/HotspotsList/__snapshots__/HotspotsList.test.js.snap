// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotspotsList component should match the snapshot when the data is has been populated 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HotspotsList
    getNearbyHotspots={[Function]}
    hotspots={
        Array [
            Object {
              "birds": Array [
                Object {
                  "comName": "Canada Goose",
                  "howMany": 10,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Branta canadensis",
                  "speciesCode": "cangoo",
                },
                Object {
                  "comName": "Graylag Goose (Domestic type) x Canada Goose (hybrid)",
                  "howMany": 1,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Anser anser (Domestic type) x Branta canadensis",
                  "speciesCode": "gragoo4",
                },
                Object {
                  "comName": "Mallard",
                  "howMany": 14,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Anas platyrhynchos",
                  "speciesCode": "mallar3",
                },
                Object {
                  "comName": "Ring-billed Gull",
                  "howMany": 2,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Larus delawarensis",
                  "speciesCode": "ribgul",
                },
                Object {
                  "comName": "Double-crested Cormorant",
                  "howMany": 4,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Phalacrocorax auritus",
                  "speciesCode": "doccor",
                },
                Object {
                  "comName": "Great Blue Heron",
                  "howMany": 1,
                  "lat": 39.7044793,
                  "lng": -105.0857735,
                  "locId": "L1743685",
                  "locName": "Belmar Park",
                  "locationPrivate": false,
                  "obsDt": "2018-09-09 12:15",
                  "obsReviewed": false,
                  "obsValid": true,
                  "sciName": "Ardea herodias",
                  "speciesCode": "grbher3",
                },
              ],
              "countryCode": "US",
              "lat": 39.7044793,
              "lng": -105.0857735,
              "locID": "L1743685",
              "locId": "L1743685",
              "locName": "Belmar Park",
              "subnational1Code": "US-CO",
              "subnational2Code": "US-CO-059",
            },
          ]
    }
    loading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter()
          currentPage="Nearby Hotspots"
/>,
        <Wrapper />,
        <main
          className="main-content"
>
          <Link
                    className="hotspot"
                    replace={false}
                    to="/hotspots/L1743685"
          >
                    Belmar Park
          </Link>
</main>,
      ],
      "className": "hotspots-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentPage": "Nearby Hotspots",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="hotspot"
              replace={false}
              to="/hotspots/L1743685"
>
              Belmar Park
</Link>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "L1743685",
            "nodeType": "class",
            "props": Object {
              "children": "Belmar Park",
              "className": "hotspot",
              "replace": false,
              "to": "/hotspots/L1743685",
            },
            "ref": null,
            "rendered": "Belmar Park",
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter()
            currentPage="Nearby Hotspots"
/>,
          <Wrapper />,
          <main
            className="main-content"
>
            <Link
                        className="hotspot"
                        replace={false}
                        to="/hotspots/L1743685"
            >
                        Belmar Park
            </Link>
</main>,
        ],
        "className": "hotspots-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": "Nearby Hotspots",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="hotspot"
                replace={false}
                to="/hotspots/L1743685"
>
                Belmar Park
</Link>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "L1743685",
              "nodeType": "class",
              "props": Object {
                "children": "Belmar Park",
                "className": "hotspot",
                "replace": false,
                "to": "/hotspots/L1743685",
              },
              "ref": null,
              "rendered": "Belmar Park",
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`HotspotsList component should match the snapshot when the data is still loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HotspotsList
    getNearbyHotspots={[Function]}
    hotspots={Array []}
    loading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter()
          currentPage="Nearby Hotspots"
/>,
        false,
        <main
          className="main-content"
>
          <LoadingSpinner />
</main>,
      ],
      "className": "hotspots-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentPage": "Nearby Hotspots",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <LoadingSpinner />,
          "className": "main-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter()
            currentPage="Nearby Hotspots"
/>,
          false,
          <main
            className="main-content"
>
            <LoadingSpinner />
</main>,
        ],
        "className": "hotspots-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": "Nearby Hotspots",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LoadingSpinner />,
            "className": "main-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
